### PR TITLE
added 8th HW

### PR DESCRIPTION
**Pull request by 415816**
Added 8th HW

### CODE REVIEW

File: lec_8_HW/task_2.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+// 2. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function getEnterpriseName(identifier) {
+    for (const enterprise of enterprises) {
+        for (const department of enterprise.departments) {
+            if (department.id === identifier || department.name === identifier) {
+                return enterprise.name;
+            }
+        }
+    }
+    return null;
```
Comment (anatoly-karpovich) [inline lec_8_HW/task_2.js]: throw new Error

File: lec_8_HW/task_2.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+}
+
+// console.log(getEnterpriseName(4));
+// console.log(getEnterpriseName("Отдел маркетинга"));
+
+// 3. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function addEnterprise(name) {
+    const newId = enterprises.reduce((maxId, enterprise) => Math.max(maxId, enterprise.id), 0) + 1;
```
Comment (anatoly-karpovich) [inline lec_8_HW/task_2.js]: need to create function for new id creation among all enterprises and departments

File: lec_8_HW/task_2.js [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+// addEnterprise("NewEnterprise_1");
+// console.log(enterprises);
+
+// 4. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function addDepartment(enterpriseId, departmentName) {
+    const enterprise = enterprises.find(enterprise => enterprise.id === enterpriseId);
+    if (enterprise) {
+        const newId =
+            enterprises.flatMap(e => e.departments).reduce((maxId, department) => Math.max(maxId, department.id), 0) +
```
Comment (anatoly-karpovich) [inline lec_8_HW/task_2.js]: Same, use the only get new id function

File: lec_8_HW/task_2.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+// deleteDepartment(10);
+// console.log(enterprises);
+
+// 9. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function moveEmployees(fromDepartmentId, toDepartmentId) {
+    const fromDepartment = enterprises
+        .flatMap(e => e.departments)
+        .find(department => department.id === fromDepartmentId);
+    const toDepartment = enterprises.flatMap(e => e.departments).find(department => department.id === toDepartmentId);
```
Comment (anatoly-karpovich) [inline lec_8_HW/task_2.js]: Duplicate of flatmap action. Set result of flatmap into a variable and then use it to find everything u need

**Review by anatoly-karpovich — APPROVED** [review]

File: lec_8_HW/task_2.js [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+    if (fromDepartment && toDepartment) {
```
Comment (VladimirLQA) [inline lec_8_HW/task_2.js]: Add check whether two departments belong to the same enterprise

**Review by VladimirLQA — APPROVED** [review]

File: lec_8_HW/task_2.js (hunk 1 above)
Comment (415816) [reply lec_8_HW/task_2.js]: accepted, code corrected

File: lec_8_HW/task_2.js (hunk 2 above)
Comment (415816) [reply lec_8_HW/task_2.js]: accepted, code corrected

File: lec_8_HW/task_2.js (hunk 3 above)
Comment (415816) [reply lec_8_HW/task_2.js]: accepted, code corrected

File: lec_8_HW/task_2.js (hunk 4 above)
Comment (415816) [reply lec_8_HW/task_2.js]: accepted, code corrected

File: lec_8_HW/task_2.js (hunk 5 above)
Comment (415816) [reply lec_8_HW/task_2.js]: accepted, code corrected

**Pull request merged**